### ISSUE TITLE
rule 'require-namespace' from policy 'disallow-default-namespace' does not work as expected

### ISSUE THREAD

**Issue by shilpa-0** [kyverno/policies#90]
Hi,

Do we really need the rule require-namespace in the policy https://github.com/kyverno/policies/blob/main/best-practices/disallow_default_namespace/disallow_default_namespace.yaml?

I expected the below pod to fail the rule but that is not the case. 
```
apiVersion: v1
kind: Pod
metadata:
  name: pod-no-namespace
  labels:
    app: myapp
spec: 
  containers:
  - name: nginx
    image: nginx
```

Is the rule indeed not working as expected or is my understanding wrong here? 

Regards,
Shilpa

**Comment by JimBugwadia** [comment]
@shilpa-0 - can you try setting the `validationFailureAction: audit` to `enforce`? 

See: https://kyverno.io/docs/writing-policies/validate/#validation-failure-action for more details.

**Comment by shilpa-0** [comment]
@JimBugwadia The behaviour is the same even with `validationFailureAction` set to `enforce`

**Comment by JimBugwadia** [comment]
Here is what I see with this policy after changing the validationFailureAction to enforce:

```sh
kubectl get cpol -A
NAME                         BACKGROUND   ACTION
disallow-default-namespace   true         enforce
```
```sh
kubectl run nginx --image=nginx
Error from server: admission webhook "validate.kyverno.svc" denied the request:

resource Pod/default/nginx was blocked due to the following policies

disallow-default-namespace:
  validate-namespace: 'validation error: Using ''default'' namespace is not allowed.
    Rule validate-namespace failed at path /metadata/namespace/'
```

Can you provide your policy settings and versions?

**Comment by shilpa-0** [comment]
Kyverno version: v1.4.1
Policy: same as https://github.com/kyverno/policies/blob/main/best-practices/disallow_default_namespace/disallow_default_namespace.yaml

Manual testing:
I switched the context to a non-default namespace and the pod creation was not blocked even though namespace tag was not mentioned.
When I switched to default namespace, the pod creation was blocked as expected. 

Using Kyverno CLI:
A pod with no `namespace` specified does not fail the rule `require-namespace` in the policy.
